.github/workflows/desktop-build.yml

```diff
@@ -230,10 +230,10 @@ jobs:
       fail-fast: false
       matrix:
         include:
-          - target: 13
+          - target: 14
             soc: Intel
-            os: macos-13
-            xcode: "14.3.1"
+            os: macos-14
```

Comment:
To my understanding, the resulting build will be targeting ARM as that's the base image.
The `soc` value is only used to add a label to the resulting build to differentiale Intel/ARM as we had both before.
We can potentially get rid off it completely if we are now ARM-only for Mac builds, and just put a note in the release notes that we no longer support macOS 13, as well as Intel-based Mac's.

<br>

@fwcd mentioned in https://github.com/Cockatrice/Cockatrice/pull/5033#issuecomment-2216106032 some time ago that using the newer images you can still target older macOS versions. We never looked into it.

Not sure the same is possible for arch.

Reply:
interesting, but we still depend on homebrew to give us deps, if github doesn't have a runner that is factually supported by homebrew I don't know how we can build at all for x64/intel, unless it's also possible to tell homebrew to install x64 packages?

Reply:
My understanding is right now that both macOS 14 binaries here are similar and both are targeting ARM arch. So they would not run on Intel Macs.

If somebody can validate that assumption with the artifacts build in this PR, that would be a good insight already!

It should also not only be about homebrew but also Xcode and the actual arch this is build on. Not sure about cross-os compilation on ARM arch or emulation to build for Intel targets? 🤔

Reply:
their file sizes are identical and the documentation clearly states that this is an arm runner, I don't think we need to test this, this is clearly an arm image, our only options are deprecating intel mac support or figuring out cross compilation

Reply:
There are [Universal Binaries](https://developer.apple.com/documentation/apple-silicon/building-a-universal-macos-binary) that combine both variants of the code. Xcode 12.2 and later support it.
I doubt that without additional configuration, the Intel one is included still and such a universal one is build? The docs mention that ARM is automatically added though.

<br>

If we can not figure out cross-compilation or the universal binary, there should be a note in the next release (similar to macOS 10.13 deprecation).